assets/js/modules/reader-revenue-manager/datastore/publications.js

```diff
@@ -68,35 +68,38 @@ const baseActions = {
 	 */
 	*syncPublicationOnboardingState() {
 		const registry = yield commonActions.getRegistry();
-		const connected = yield commonActions.await(
-			registry
-				.resolveSelect( CORE_MODULES )
-				.isModuleConnected( READER_REVENUE_MANAGER_MODULE_SLUG )
-		);
 
-		// If the module is not connected, do not attempt to sync the onboarding state.
-		if ( ! connected ) {
+		const hasPublicationIDChanged = registry
+			.select( MODULES_READER_REVENUE_MANAGER )
+			.hasSettingChanged( 'publicationID' );
+
+		// Do not attempt to sync the onboarding state if the publication ID
+		// in state is not the "saved" publication ID.
+		if ( hasPublicationIDChanged ) {
 			return;
 		}
```

Comment:
As the sync is done on the client-side, there is a possibility where we may set an incorrect onboarding state in settings in case the publication ID has been changed in store. To avoid this, I've added this check so that we only attempt the sync when the publication ID has not changed from its saved value.